### ISSUE TITLE
Ruff doesn't raise syntax error for single starred assignment target

### ISSUE THREAD

**Issue by dhruvmanila** [astral-sh/ruff#13759]
We should be raising syntax error for the following code:

```py
*a = (1,)
```

https://play.ruff.rs/7e3fe1a8-7edc-40eb-a415-ced8f61bb7c1

**Comment by dhruvmanila** [comment]
Oh, I think this is caught by the compiler and not the parser as `python -m ast test.py` passes without any syntax errors. So, this is a part of #11934.
